crate/memories/examples/postgresql.rs

```diff
@@ -0,0 +1,109 @@
+//! This example show-cases the use of Findex to securely store a hash-map with postgresql.
+#[path = "shared_utils.rs"]
+mod shared_utils;
+
+use cosmian_crypto_core::{CsRng, Secret, reexport::rand_core::SeedableRng};
+use cosmian_findex_memories::{
+    PostgresMemory, PostgresMemoryError,
+    reexport::{
+        cosmian_findex::{ADDRESS_LENGTH, Address, Findex, IndexADT, MemoryEncryptionLayer},
+        deadpool_postgres::{Config, Pool},
+        tokio_postgres::NoTls,
+    },
+};
+use futures::executor::block_on;
+use shared_utils::{WORD_LENGTH, decoder, encoder, gen_index};
+use std::collections::HashMap;
+
+const DB_URL: &str = "postgres://cosmian:cosmian@localhost/cosmian";
+const TABLE_NAME: &str = "findex_example";
+
+async fn create_pool(db_url: &str) -> Result<Pool, PostgresMemoryError> {
+    let mut pg_config = Config::new();
+    pg_config.url = Some(db_url.to_string());
+    let pool = pg_config.builder(NoTls)?.build()?;
+    Ok(pool)
+}
+
+#[tokio::main]
+async fn main() {
+    // For cryptographic applications, it is important to use a secure RNG. In
+    // Rust, those RNG implement the `CryptoRng` trait.
+    let mut rng = CsRng::from_entropy();
+
+    // Generate fresh Findex key. In practice only one user is in charge of
+    // generating the key (the administrator?): all users *must* share the same
+    // key in order to make the index inter-operable.
+    let key = Secret::random(&mut rng);
+
+    // Generating the random index.
+    let index = gen_index(&mut rng);
+
+    let pool = create_pool(DB_URL).await.unwrap();
+    let m = PostgresMemory::<Address<ADDRESS_LENGTH>, [u8; WORD_LENGTH]>::connect_with_pool(
+        pool.clone(),
+        TABLE_NAME.to_string(),
+    )
+    .await
+    .unwrap();
+
+    // Notice we chose to not enable Tls: it's not needed for this example as we are
+    // using the encryption layer in top of the memory interface - i.e. the data is
+    // already encrypted before being sent to the database and Tls would add unnecessary overhead.
```

Comment:
Capitalize acronyms adequately:
```suggestion
    // Notice we chose to not enable TLS: it's not needed for this example as we
    // are using the encryption layer in top of the memory interface - i.e. the
    // data is already encrypted before being sent to the database and TLS would
    // add unnecessary overhead.
```